### PR TITLE
Add first implementation of CE mixin

### PR DESCRIPTION
**Pull request by teojgo**
https://jira.cscs.ch/browse/VCUE-217

### CODE REVIEW

**Review by ekouts — COMMENTED** [review]
Looks good! I have only one small comment, add the `ce` feature in https://github.com/eth-cscs/cscs-reframe-tests/blob/alps/portable_tests.md . I am not sure it's the best place, but we have to have a list/database with the features and what they mean.

**Review by ekouts — APPROVED** [review]
lgtm

File: checks/containers/container_engine/check_cuda_nbody.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        curr_part = self.current_partition
+        self.num_gpus_per_node = curr_part.select_devices('gpu')[0].num_devices
+
+    @run_after('setup')
+    def set_executable_opts(self):
+        self.executable_opts = [
+            f'-benchmark', f'-fp64',
+            f'-numbodies={self.num_bodies_per_gpu * self.num_gpus_per_node}',
+            f'-numdevices={self.num_gpus_per_node}'
+        ]
```
Comment (jgphpc) [inline checks/containers/container_engine/check_cuda_nbody.py:45]:
```suggestion
        ]
```

File: portable_tests.md [hunk 2]
```diff
@@ -124,6 +126,18 @@ class MemBandwidthTest(rfm.RunOnlyRegressionTest):
 Currently, ReFrame supports auto-detection of the local or remote processor information only.
 It does not support auto-detection of devices, in which cases users should explicitly specify this information using the `devices` [configuration option](https://reframe-hpc.readthedocs.io/en/stable/config_reference.html#config.systems.partitions.devices).
 
+## How to run Container Engine(CE) based tests
+
+To run a Container Engine (CE) based test, it's essential to create a test class inheriting from the `ContainerEngineMixin` and add the `ce` as one of the `valid_system` features.
```
Comment (jgphpc) [inline portable_tests.md:131]:
```suggestion
To run a Container Engine (CE) based test, it is essential to create a test class inheriting from the `ContainerEngineMixin` and add the `ce` as one of the `valid_system` features. 
```

File: checks/containers/container_engine/check_cuda_nbody.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
+# Copyright 2016-2023 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
```
Comment (jgphpc) [inline checks/containers/container_engine/check_cuda_nbody.py:1]: since it's  a new test, the copyright year should be 2024 only

Comment (jgphpc) [reply checks/containers/container_engine/check_cuda_nbody.py:1]:
```suggestion
# Copyright 2024 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
```

**Pull request merged**